src/data/datasets/detection.py

```diff
@@ -0,0 +1,136 @@
+                    min_area: float = 0.0,
+                    min_visibility: float = 0.0,
+                    **dataset_params
+    ):
+        super().__init__(**dataset_params)
+        
+        self.target_column = target_column
+        if self.augment is not None:
+            self.augment = module_transforms.Compose(
+                self.augment,
+                bbox_params=module_transforms.BboxParams(
+                    format='pascal_voc',
+                    label_fields=['category_ids'],
+                    min_area=min_area,
+                    min_visibility=min_visibility
+                    )
+            )
+
+        self.transform = module_transforms.Compose(
+                self.transform,
+                bbox_params=module_transforms.BboxParams(
+                    format='pascal_voc',
+                    label_fields=['category_ids'],
+                    min_area=min_area,
+                    min_visibility=min_visibility
+                    )
+            )
+
+        self.csv[target_column] = self.csv[target_column].apply(eval)
+
+    def __getitem__(self, idx: int):
+        sample = self.get_raw(idx // self.expand_rate)
+        sample['image'] = sample['image'].type(torch.__dict__[self.input_dtype])
+        
+        output = {
+            'input': sample['image'],
+            'target_bboxes': torch.tensor(sample['bboxes']).type(torch.__dict__[self.target_dtype]),
+            'target_labels': torch.tensor(sample['category_ids']).type(torch.__dict__[self.target_dtype]),
+            'bbox_count': torch.tensor(sample['bbox_count'])
+        }
+
+        return output
+        
+    def get_raw(self, idx: int):
+        record = self.csv.iloc[idx]
+        image = self.read_image(record)
+        row_annotations = record[self.target_column]
+
+        bboxes = []
+        labels = []
+        for annotation in row_annotations:
+            bbox = [annotation['x_min'], annotation['y_min'], annotation['x_max'], annotation['y_max']]
+            label = annotation['label']
+            bboxes.append(bbox)
+            labels.append(label)
+
+        bbox_count = len(bboxes)
+
+        sample = {
+            'image': image,
+            'bboxes': bboxes,
+            'category_ids': labels
+            }
+
+        if self.augment is not None:
+            sample = self.augment(**sample)
+
+        sample = self.transform(**sample)
+        sample['bbox_count'] = bbox_count
+        
+        return sample
+
+    @staticmethod
+    def collate_fn(batch: dict) -> dict:
+        """
+        Add empty bbox and label into batch with different size of bboxes
+        empty bbox = [0, 0, 0, 0]
+        empty label = -1
+        """
+        # get sequence lengths
+        max_length = 0
+        for t in batch:
+            max_length = max(max_length, t['bbox_count'])
+       
+        empty_box = [0]*4
+        for t in batch:
+            bbox_count = t['bbox_count']
```

Comment:
Make it simple and faster:
```
bboxes = torch.zeros(max_length, 4, dtype=torch.float16)
bboxes[:t['bbox_count']] = t['target_bboxes']
labels = torch.full(max_length, value=-1, dtype=torch.long)
labels[:t['bbox_count']] = t['target_labels']
```

Reply:
about float16, we have x_min, y_min, x_max, y_max format, we dont care about long and change into float16?